questions.md

```diff
@@ -0,0 +1,9 @@
+#Questions:
+
+1. If i want to make a pull request to repo of other user, do i need to clone it first?
+2. In the classroom every studen has a repo, they need to make a pull to download all and create a new branch and stat to work on it and then just push ?
+3. Really confused, should i do aways push from the branch master to the origin ?
```

Comment:
If you are working on the `master` branch, then yes, you should push from that branch. _However_, it is not commonly recommended to work on `master`. The ideal setup is this:

- Every repository has a **default** branch. In git, these branches are called `master` by default, but that can be changed. When we discuss this, I will use `master` to refer to the default branch.
- The `master` branch should be considered the most complete, pristine copy of a repository. In large-scale software projects, _nothing_ should be merged to master without thorough testing and review. This is the habit we are trying to build in our students when we have them work on "working" branches.
- So no, in an ideal world you do not push to `master`, you push from a working branch and _merge_ into `master` using a pull request.
- Of course, don't make things more difficult then they need to be, if you are just adding instructions to a README for your student's starter repository, there's little reason not to just push from `master`, the flow is adaptable to whatever works for you!